### PR TITLE
SES6: Maintenance QA update testing

### PR DESCRIPTION
**Pull request by shyukri**
Clone of PR #368 for SES6.

Test run: https://storage-ci.suse.de/job/storage-maintenance-update-6.0/21/ targeting https://smelt.suse.de/incident/14301/ patch

### CODE REVIEW

**Comment by smithfarm** [comment]
@shyukri Sorry, I had to revert `ses6-downstream-commits` to the currently shipping SES6 code. Can you please `git reset --hard` and `git cherry-pick` your commit and force push?

**Comment by shyukri** [comment]
Done

**Comment by smithfarm** [comment]
I noticed #368 is marked DNM. Does that mean this one is DNM too?

**Comment by shyukri** [comment]
Latest run: https://storage-ci.suse.de/job/storage-run-teuthology-mu-patch-6.0/21/
On #368 i see some issues on the iscsi testing part which will require further debug.
This one shouldn't be DNM.

File: qa/tasks/deepsea.py [hunk 1]
```diff
@@ -1465,7 +1467,28 @@ def begin(self):
...
+            paramdict = {'tries': reboot_tries}
+        paramdict_keys = len(paramdict)
+        if paramdict_keys >= 1:
+            self.log.info("Considering parameter dict {}".format(paramdict))
+            if paramdict_keys > 1:
+                raise ConfigError(
+                    self.err_prefix +
+                    "parameter dict may contain only one key. "
+                    "You provided ->{}<- keys".format(paramdict_keys)
+                    )
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1481]: Why not to move this check before `paramdict_keys >= 1`?

Comment (kshtsk) [reply qa/tasks/deepsea.py:1481]: and use later `elif paramdict_keys == 1`

File: qa/tasks/deepsea.py [hunk 2]
```diff
@@ -1465,7 +1467,28 @@ def begin(self):
...
+            tries, tries_num = paramdict.items()[0]
+            if tries != 'tries':
+                raise ConfigError(
+                    self.err_prefix +
+                    "The parameter dict only supports one key: \"tries\""
+                    "But you provided an unknown key ->{}<-".format(tries)
+                    )
+            tries_num = int(tries_num)
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1489]: I suggest to use straight logic here:
```
if tries == 'tries':
  tries_num = int(tries_num)
else:
  raise ...
```

File: qa/tasks/deepsea.py [hunk 3]
```diff
@@ -1465,7 +1467,28 @@ def begin(self):
                 "config dictionary may contain only one key. "
                 "You provided ->{}<- keys ({})".format(len(config_keys), config_keys)
                 )
-        role_spec, repositories = self.config.items()[0]
+        role_spec, paramdict = self.config.items()[0]
+        if not isinstance(paramdict, dict):
+            paramdict = {'tries': reboot_tries}
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1472]: Is it a common practice to override parameters if they have different type? what if it is number value and user will not be aware that he entered something wrong.

File: qa/tasks/deepsea.py (hunk 2 above)
Comment (shyukri) [reply qa/tasks/deepsea.py]: Done

File: qa/tasks/deepsea.py (hunk 1 above)
Comment (shyukri) [reply qa/tasks/deepsea.py]: Done

File: qa/tasks/deepsea.py (hunk 3 above)
Comment (shyukri) [reply qa/tasks/deepsea.py:1472]: `reboot_tries` is already defined on top of the file

**Comment by gekios** [comment]
lgtm too :)

**Pull request merged**